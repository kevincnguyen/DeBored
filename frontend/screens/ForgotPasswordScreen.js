const ForgotPassordScreen = () => {
  return null;
};

export default ForgotPassordScreen;
