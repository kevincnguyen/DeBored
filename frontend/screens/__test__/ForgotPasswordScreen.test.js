import React from "react";
import renderer from "react-test-renderer";
import ForgotPasswordScreen from "../ForgotPasswordScreen";

// Mock any necessary modules or components
jest.mock("react-native-paper", () => ({
  Button: "Button",
  TextInput: "TextInput",
}));

test("renders ForgotPasswordScreen component correctly", () => {
  const tree = renderer.create(<ForgotPasswordScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("contains TextInput for email", () => {
  const component = renderer.create(<ForgotPasswordScreen />);
  const textInput = component.root.findByProps({ label: "Email" });
  expect(textInput).toBeTruthy();
});

test("contains a button to send email", () => {
  const component = renderer.create(<ForgotPasswordScreen />);
  const button = component.root.findByType("Button");
  expect(button).toBeTruthy();
  // Check button text if possible
});
