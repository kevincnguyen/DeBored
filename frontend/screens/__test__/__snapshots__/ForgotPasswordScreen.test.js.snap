// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ForgotPasswordScreen component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
      "paddingBottom": 150,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 35,
        "fontWeight": "bold",
        "marginBottom": 35,
      }
    }
  >
    Forgot password?
  </Text>
  <View
    style={
      {
        "width": "90%",
      }
    }
  >
    <TextInput
      label="Email"
      mode="outlined"
      onChangeText={[Function]}
      value=""
    />
    <Button
      contentStyle={
        {
          "paddingVertical": 5,
        }
      }
      labelStyle={
        {
          "fontSize": 16,
        }
      }
      mode="contained"
      onPress={[Function]}
      style={
        {
          "borderRadius": 10,
          "marginTop": 50,
        }
      }
    >
      Send email
    </Button>
  </View>
</View>
`;
